pyvista/plotting/lights.py

```diff
@@ -0,0 +1,465 @@
+        positional light.
+
+        """
+        return bool(self.GetPositional())
+
+    @positional.setter
+    def positional(self, state):
+        """Set whether the light should be positional."""
+        self.SetPositional(state)
+
+    @property
+    def exponent(self):
+        """Return the exponent of the cosine used in positional lighting."""
+        return self.GetExponent()
+
+    @exponent.setter
+    def exponent(self, exp):
+        """Set the exponent of the cosine used in positional lighting."""
+        self.SetExponent(exp)
+
+    @property
+    def cone_angle(self):
+        """Return the cone angle of a positional light.
+
+        The angle is in degrees and is measured between the axis of the cone
+        and an extremal ray of the cone. A value smaller than 90 has spot
+        lighting effects, anything equal to and above 90 is just a positional
+        light.
+
+        """
+        return self.GetConeAngle()
+
+    @cone_angle.setter
+    def cone_angle(self, angle):
+        """Set the cone angle of a positional light."""
+        self.SetConeAngle(angle)
+
+    @property
+    def attenuation_values(self):
+        """Return the quadratic attenuation constants.
+
+        The values specify the constant, linear and quadratic constants
+        in this order.
+
+        """
+        return self.GetAttenuationValues()
+
+    @attenuation_values.setter
+    def attenuation_values(self, values):
+        """Set the quadratic attenuation constants."""
+        self.SetAttenuationValues(values)
+
+    # TODO: implement transformation_matrix here?
+
+    @property
+    def light_type(self):
+        """Return the light type.
+
+        The default light type is a scene light which lives in world
+        coordinate space.
+
+        A headlight is attached to the camera and always points at the
+        camera's focal point.
+
+        A camera light also moves with the camera, but it can have an
+        arbitrary relative position to the camera. Camera lights are
+        defined in a coordinate space where the camera is located at
+        (0, 0, 1), looking towards (0, 0, 0) at a distance of 1, with
+        up being (0, 1, 0). Camera lights use the transform matrix to
+        establish this space.
+
+        The property returns class constant values from an enum:
+            - Light.HEADLIGHT == 1
+            - Light.CAMERA_LIGHT == 2
+            - Light.SCENE_LIGHT == 3
+
+        """
+        return LightType(self.GetLightType())
+
+    @light_type.setter
+    def light_type(self, ltype):
+        """Set the light type.
+
+        Either an integer code or a class constant enum value must be used.
+
+        """
+        if not isinstance(ltype, int):
+            raise TypeError('Light type must be an integer subclass,'
+                            f' got {ltype} instead.')
+        # LightType is an int subclass; convert to proper int
+        ltype = int(ltype)
+        self.SetLightType(ltype)
```

Comment:
Comment isn't super helpful.

```suggestion
        self.SetLightType(int(ltype))
```

Reply:
I kind of agree, but I wanted to make it clear that `LightType` values are covered by the `isinstance` check. If you think that's trivial then no comment is needed. (Or I'm open to suggestions for a more helpful comment.)

But now I realize that exactly because of the `int` subtype we can directly feed the type to `SetLightType`, so the `int(ltype)` call itself is superfluous.

Reply:
Specifying that LightType is accepted in the docstring might be more helpful, and if vtk accepts it without casting then it doesn't matter IMO.

Reply:
I'm mostly worried about devs having to spend mental CPU cycles on figuring out that the `int` case covers `LightType`. The docstring is more for end users in my head. But anyway I've mentioned these enum values in the `Light` class docstring. When I get to writing the proper documentation and examples I'll take care to make this clear.

But you're probably right about it not being an issue without an explicit `int` conversion. Thanks.